### PR TITLE
Add dependency graph generation & auto merge of dependency PRs

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
Create ballerina version bump dependency PRs as per the dependency graph and not in alphabetical order
Auto merge Dependency bump PRs

Fixes https://github.com/ballerina-platform/ballerina-release/issues/131

## Approach
1. Create dependency graph using `networkx` library
2. Approve opened dependency PRs with `ballerina-reviewer-bot`
3. Merge PRs once the build checks pass (Build checks is checked for 40 minutes in every 30s interval)

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: dependabot/update_lang_version.py [hunk 1]
```diff
@@ -82,26 +93,119 @@ def get_module_list_json():
...
+        else:
+            modules_list[index][MODULE_PR_CHECK_STATUS] = "failure"
+            modules_list[index][MODULE_FAILED_PR_CHECKS] = failed_pr_checks
+            pr_failed_modules.append(modules_list[index])
+        del(modules_list[index])
+
+def update_module(github, module, lang_version):
+    repo = github.get_repo(ORGANIZATION + "/" + module[MODULE_NAME])
     try:
         properties_file = repo.get_contents(PROPERTIES_FILE, ref=LANG_VERSION_UPDATE_BRANCH)
```
Comment (manuranga) [inline dependabot/update_lang_version.py:200]: shouldn't we always pick up from master?

Comment (niveathika) [reply dependabot/update_lang_version.py]: Fixed in 74ab50a

File: dependabot/update_version_data.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+            dependencies.append(module[:-2])
+
+    return dependencies
+
+# Gets the version of the ballerina extension module from gradle.properties file in module repository
+# returns: current version of the module
+def get_version(module_name):
+    try:
+        data = url_open_with_retry("https://raw.githubusercontent.com/" + BALLERINA_ORG_NAME + "/"
+                                    + module_name + "/master/gradle.properties")
```
Comment (keizer619) [inline dependabot/update_version_data.py:106]: @niveathika Can this be `main` in new repos? Just wondering whether should we parameterize to support release from other branches?

Comment (manuranga) [reply dependabot/update_version_data.py:106]: And why is this not happing via the githuv api like eveyting else?

Comment (niveathika) [reply dependabot/update_version_data.py]: @keizer619 Fixed it in 3e0a99a
@manuranga It was from legacy code, I have updated it.

**Review by keizer619 — APPROVED** [review]

**Pull request merged**